### PR TITLE
Fixes on providers

### PR DESCRIPTION
**Pull request by Stormunblessed**
Fixes on Cinecalidad and added vpn might be needed

### CODE REVIEW

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/CinecalidadProvider.kt [hunk 1]
```diff
@@ -106,13 +95,15 @@ class CinecalidadProvider : MainAPI() {
...
         val title = soup.selectFirst(".single_left h1").text()
-        val description =
-            soup.selectFirst(".single_left > table:nth-child(3) > tbody:nth-child(1) > tr:nth-child(1) > td:nth-child(2) > p")
-                ?.text()?.trim()
+        val description = soup.selectFirst("div.single_left table tbody tr td p")?.text()?.trim()
         val poster: String? = soup.selectFirst(".alignnone").attr("data-src")
         val episodes = soup.select("div.se-c div.se-a ul.episodios li").map { li ->
             val href = li.selectFirst("a").attr("href")
-            val epThumb = li.selectFirst("div.imagen img").attr("data-src")
+            val epThumb = try {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/CinecalidadProvider.kt:102]: you could use ?: instead of try catch

File: app/src/main/java/com/lagradost/cloudstream3/utils/ExtractorApi.kt [hunk 2]
```diff
@@ -11,7 +11,7 @@ import org.jsoup.Jsoup
 
 data class ExtractorLink(
     val source: String,
-    val name: String,
+    var name: String,
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/utils/ExtractorApi.kt:14]: undo this change, there is no need for a name that can be changed

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/CinecalidadProvider.kt [hunk 3]
```diff
@@ -152,33 +142,33 @@ class CinecalidadProvider : MainAPI() {
...
-                Regex("(https?:\\/\\/(www\\.)?[-a-zA-Z0-9@:%._\\+~#=]{1,256}\\.[a-zA-Z0-9()]{1,6}\\b([-a-zA-Z0-9()@:%_\\+.~#?&\\/\\/=]*))")
-            val links = serversRegex.findAll(videos.toString()).map { it.value }.toList()
-            for (link in links) {
-                for (extractor in extractorApis) {
-                    if (link.startsWith(extractor.mainUrl)) {
-                        extractor.getSafeUrl(link, data)?.forEach {
-                            callback(it)
-                        }
-                    }
+        app.get(data).document.select(".dooplay_player_option").forEach {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/CinecalidadProvider.kt:151]: you can use .apmap instead of forEach to make all requests in parallel

**Review by LagradOst — CHANGES_REQUESTED** [review]
Looks, good, however you need to undo the change on extractorapi as it serves no purpose. The rest are just general feedback on the code

**Comment by Stormunblessed** [comment]
I think that's it

**Comment by LagradOst** [comment]
LGTM, ready to merge?

**Comment by Stormunblessed** [comment]
> 

Yeah
